Add support for serializing a RegExp to JSON (#521)
stdlib-js/stdlib: lib/node_modules/@stdlib/regexp/to-json/docs/repl.txt

{{alias}}( regexp )
    Returns a JSON representation of a regular expression.

    The returned object has the following properties:

    - type: value type. The assigned value is always 'RegExp'.
    - pattern: regular expression pattern.
    - flags: regular expression flags.

    Parameters
    ----------
    regexp: RegExp
        Regular expression to serialize.

    Returns
    -------

>     out: Object
>         JSON representation.
> 
>     Examples

Reviewer: ```suggestion
    out: Object
        JSON representation.
        
    out.type: string
        Value type.
        
    out.pattern: string
        Regular expression pattern.
        
    out.flags: string
        Regular expression flags.

    Examples
```

    --------
    > var json = {{alias}}( /ab+c/ )

>     <Object>

Reviewer: ```suggestion
    {...}
```

Instead of `<Object>` here, we typically use `{...}` when the return value is a non-empty plain object.

    See Also

>     --------

Reviewer: ```suggestion
    --------
    
```

I think we're missing a blank line here. Should have 2 line breaks after the See Also section.

{{alias}}( regexp )

>     Returns a JSON representation of a RegExp object.

Reviewer: ```suggestion
    Returns a JSON representation of a regular expression.
```

>     The JSON object is guaranteed to have the following properties:

Reviewer: ```suggestion
    The returned object has the following properties:
```

    - type: RegExp type.
    - pattern: regular expression pattern string.

>     - flags: aditional options for matching.

Reviewer: ```suggestion
    - type: value type. The assigned value is always 'RegExp'.
    - pattern: regular expression pattern.
    - flags: regular expression flags.
```

Updating to align with the README.

    Parameters
    ----------
    regexp: RegExp
        Regular expression to serialize.

    Returns
    -------
    out: Object
        JSON representation.

    Examples
    --------
    > var pattern = 'ab+c';
    > var regexp = new RegExp( pattern );

>     > var json = {{alias}}( regexp )

Reviewer: ```suggestion
    > var json = {{alias}}( /ab+c/ )
```

{{alias}}( error )
    Returns a JSON representation of a regexp object.

    The JSON object is guaranteed to have the following properties:

    - type: regexp type.
    - pattern: regular expression pattern string.

   Depending on the object, the following properties *may* be present:

    - flags: aditional options for matching.

    The function also serializes all enumerable properties.

    Parameters
    ----------
    regexp: RegExp
        Regular expression to serialize.

    Returns
    -------
    out: Object
        JSON representation.

    Examples
    --------

>     > var regex = new RegExp('ab+c');

Reviewer: In general, here and elsewhere, we include spaces around function arguments. I.e., `new RegExp( 'ab+c' );`.
